src/PartialModelSelectionMain.cpp

```diff
@@ -76,24 +76,24 @@ TEST(ModelCreationTests, DISABLED_modelLossTestPos){
 
    
   TEST(InsertTests, testLowerBoundInsertion){
-    ModelSelectionMap testMap = ModelSelectionMap();
+    ModelSelectionMap testMap = ModelSelectionMap(); //Create a new model selection map class
     Model model3Seg = Model(3, 0.0); //Previous
     Model model2Seg = Model(2, 4.0); //Given with lower_bound
-    testMap.insert(2.0, model3Seg);
-    auto lowerBeforeInserts = testMap.penaltyModelMap.lower_bound(5.0);
+    testMap.insert(2.0, model3Seg); //Insert the three segment model at penalty 3.0.
+    auto lowerBeforeInserts = testMap.penaltyModelMap.lower_bound(3.0); //This returns map::end as no key exceeds 3.0
     
-    ASSERT_EQ(lowerBeforeInserts, testMap.penaltyModelMap.end());
-    ASSERT_EQ(prev(lowerBeforeInserts)->first, 2.0);
-    testMap.insert(5.0, model2Seg);
-    testMap.displayMap();
-    auto lowerBAfterInserts = testMap.penaltyModelMap.lower_bound(3.0);
-    auto prevPair = prev(lowerBAfterInserts);
-    auto firstPair = prev(lowerBAfterInserts, 2);
-    int prevModelSize = prevPair->second.modelSize;
-    int lowerBoundModelSize = lowerBAfterInserts->second.modelSize;
-    ASSERT_EQ(2, lowerBoundModelSize);
-    ASSERT_EQ(3, prevModelSize);
-    ASSERT_EQ(1, firstPair->second.modelSize);
+    ASSERT_EQ(lowerBeforeInserts, testMap.penaltyModelMap.end()); //Testing that this is indeed the case. 
```

Comment:
it would be more relevant to have tests for the result of Minimize/getNewPenalty after each insertion.
the result of lower_bound is an implementation detail which is not relevant to the user

Reply:
I agree, but I am using this to aid in my development of the methods, as I needed to test how lower_bound behaved and what the returned iterators pointed at in the data structure so I could use lower_bound appropriately. This whole test is for my reference and shows what operations I need to perform to get the correct iterators within the minimize and insert methods.